lib/src/store.dart

```diff
@@ -58,22 +63,36 @@ class Store {
   /// As an example, [transformer] could be used to throttle the number of
   /// triggers this [Store] emits for state that may update extremely frequently
   /// (like scroll position).
-  Store.withTransformer(StreamTransformer<dynamic, dynamic> transformer) {
-    _streamController = new StreamController<Store>();
+  Store.withTransformer(StreamTransformer<dynamic, dynamic> transformer)
+      : _streamController = new StreamController<Store>() {
+    manageStreamController(_streamController);
 
     // apply a transform to the stream if supplied
     _stream = _streamController.stream
         .transform(transformer as StreamTransformer<Store, dynamic>)
```

Comment:
I initially attempted to align these types with their actual usage, but ran into issues with tests and decided it was not worth the effort to correct it.

Issues:
- The Throttler transformer used in tests is `StreamTransformer<dynamic, dynamic>`
- Useful transformers do not necessarily need to output Store
- nor do they need to return a broadcast stream

To be honest, I think it'd clean up the code quite a bit if this constructor was removed, but I don't know how often it is used in existing code.